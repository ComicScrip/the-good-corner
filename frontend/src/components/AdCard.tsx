type AdCardProps = {
  imgUrl: string;
  price: number;
  title: string;
  link: string;
};
export default function AdCard({ imgUrl, link, title, price }: AdCardProps) {
  return (
    <div className="ad-card-container">
      <a className="ad-card-link" href={link}>
        <img className="ad-card-image" src={imgUrl} />
        <div className="ad-card-text">
          <div className="ad-card-title">{title}</div>
          <div className="ad-card-price">{price} â‚¬</div>
        </div>
      </a>
    </div>
  );
}
