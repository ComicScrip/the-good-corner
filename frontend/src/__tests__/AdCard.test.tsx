import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import AdCard from "../components/AdCard";

describe("AdCard", () => {
  it("renders the AdCard component", () => {
    const view = render(
      <AdCard
        link="/ads/1"
        ad={{
          id: 1,
          title: "myad",
          picture: "http://img.com/i.png",
          price: 42,
        }}
      />
    );

    expect(screen.getByText(/myad/)).toBeVisible();
    expect(screen.getByText(/42/)).toBeVisible();
    expect(view.baseElement).toMatchSnapshot();
  });
});
