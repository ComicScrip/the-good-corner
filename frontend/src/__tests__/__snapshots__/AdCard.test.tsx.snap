// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdCard renders the AdCard component 1`] = `
<body>
  <div>
    <div
      class="w-[400px]"
    >
      <a
        href="/ads/1"
      >
        <div
          class="shadow-md border rounded-lg p-6 bg-white mr-3 mb-3"
        >
          <img
            class="h-[200px] w-full object-cover rounded-md"
            src="http://img.com/i.png"
          />
          <div
            class="flex justify-between pt-6"
          >
            <div
              class="ad-card-title"
            >
              myad
            </div>
            <div
              class="ad-card-price"
            >
              42
               â‚¬
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</body>
`;
